server_addon/houdini/client/ayon_houdini/plugins/publish/collect_files_for_cleaning_up.py

```diff
@@ -0,0 +1,112 @@
+import os
+
+import clique
+import pyblish.api
+from ayon_core.pipeline import AYONPyblishPluginMixin
+from ayon_houdini.api import plugin
+
+
+class CollectFilesForCleaningUp(plugin.HoudiniInstancePlugin,
+                                AYONPyblishPluginMixin):
+    """Collect Files For Cleaning Up.
+
+    This collector collects output files
+    and adds them to file remove list.
+
+    CAUTION:
+        This collector registers exported files and
+          the parent folder (if it was empty) for deletion
+          in `ExplicitCleanUp` plug-in.
+        Artists are free to change the file path in the ROP node.
+    """
+
+    # It should run after CollectFrames and Collect Render plugins,
+    # and before CollectLocalRenderInstances.
+    order = pyblish.api.CollectorOrder + 0.115
+
+    hosts = ["houdini"]
+    families = ["*"]
+    label = "Collect Files For Cleaning Up"
+    intermediate_exported_render = False
+
+    def process(self, instance):
+
+        files = []
+        staging_dirs = []
```

Comment:
I wonder if we're making our lifes easier no by setting type hints here:
```suggestion
        files: List[str] = []
        staging_dirs: List[str] = []
```
Or even:
```suggestion
        files: List[os.PathLike] = []
        staging_dirs: List[os.PathLike] = []
```

Reply:
Lets use `list[str]` because they can include something like 
```
f"{staging_dir}/{frames}"
```

Reply:
It seems my IDE doesn't like it:
![image](https://github.com/ynput/ayon-core/assets/2439881/c9bb36ed-f96b-460a-8bfd-d5ad2ff172be)

Let's make it `List[str]` then.

Reply:
meh, another reason to stop using VSCode. Nah I can't cheat on it.